src/zip.c

```diff
@@ -1022,3 +1022,19 @@ int zip_extract_stream(const char *stream, size_t size, const char *dir,
 
   return status;
 }
+
+struct zip_t *zip_open_stream(char *stream, size_t size) {
+  struct zip_t *zip = NULL;
+  zip = (struct zip_t *)calloc((size_t)1, sizeof(struct zip_t));
+  if (!zip) {
+    goto cleanup;
```

Comment:
if `zip` is `NULL` then there is nothing to cleanup, so we can return.

Reply:
OK